prepare_vscode.sh

```diff
@@ -85,13 +84,15 @@ fi
 
 setpath() {
   { set +x; } 2>/dev/null
-  echo "$( cat "${1}.json" | jq --arg 'path' "${2}" --arg 'value' "${3}" 'setpath([$path]; $value)' )" > "${1}.json"
+  jq --arg 'path' "${2}" --arg 'value' "${3}" 'setpath([$path]; $value)' "${1}.json" > "${1}.json.tmp"
```

Comment:
I would like to avoid tmp files

Reply:
It is certainly true the reading/writing from the same pipeline could unexpectedly truncate files. I will opt for a variable that will be unset after instead of a tmp file.

Reply:
This is my purposed improvement:
```bash
setpath() {
  { set +x; } 2>/dev/null
  json_tmp=$(jq --arg 'path' "${2}" --arg 'value' "${3}" 'setpath([$path]; $value)' "${1}.json")
  echo "$json_tmp" > "${1}.json" && unset "json_tmp"
  set -x
}
```
This could provide some error handling if `json_tmp` is still set we know we didn't update `"${1}.json" successfully. Otherwise I could just unset no matter what.

Reply:
It's true that `jq` will truncate a file if you do `jq "${1}.json" > "${1}.json"`. But it doesn't with the `echo "$( cat "${1}.json" | jq )" > "${1}.json"`

To avoid `unset`,  you can use `local jsonTmp`